dbms/src/main/java/org/polypheny/db/processing/AbstractQueryProcessor.java

```diff
@@ -135,90 +141,147 @@
 import org.polypheny.db.transaction.TableAccessMap.Mode;
 import org.polypheny.db.transaction.TableAccessMap.TableIdentifier;
 import org.polypheny.db.transaction.TransactionImpl;
-import org.polypheny.db.type.ArrayType;
-import org.polypheny.db.type.ExtraPolyTypes;
 import org.polypheny.db.type.PolyType;
 import org.polypheny.db.util.ImmutableIntList;
 import org.polypheny.db.util.Pair;
-import org.polypheny.db.util.Util;
 import org.polypheny.db.view.MaterializedViewManager;
 import org.polypheny.db.view.MaterializedViewManager.TableUpdateVisitor;
 import org.polypheny.db.view.ViewManager.ViewVisitor;
 
 
 @Slf4j
-public abstract class AbstractQueryProcessor implements QueryProcessor {
+public abstract class AbstractQueryProcessor implements QueryProcessor, ExecutionTimeObserver {
 
     protected static final boolean ENABLE_BINDABLE = false;
     protected static final boolean ENABLE_COLLATION_TRAIT = true;
     protected static final boolean ENABLE_ENUMERABLE = true;
     protected static final boolean CONSTANT_REDUCTION = false;
     protected static final boolean ENABLE_STREAM = true;
-
     private final Statement statement;
 
+    private final CachedPlanRouter cachedPlanRouter = RoutingManager.getInstance().getCachedPlanRouter();
+    private final RoutingPlanSelector routingPlanSelector = RoutingManager.getInstance().getRoutingPlanSelector();
+    private final RoutingDebugUiPrinter debugUiPrinter = RoutingManager.getInstance().getDebugUiPrinter();
+    private final DmlRouter dmlRouter = RoutingManager.getInstance().getDmlRouter();
+
+    private final Map<Integer, Long> scanPerTable = new HashMap<>(); // scanId  -> tableId //Needed for Lookup
+
 
     protected AbstractQueryProcessor( Statement statement ) {
         this.statement = statement;
     }
 
 
     @Override
-    public PolyphenyDbSignature prepareQuery( RelRoot logicalRoot ) {
-        return prepareQuery( logicalRoot, logicalRoot.rel.getCluster().getTypeFactory().builder().build(), false );
+    public void executionTime( String reference, long nanoTime ) {
+        StatementEvent event = statement.getTransaction().getMonitoringEvent();
+        if ( reference.equals( event.getLogicalQueryInformation().getQueryClass() ) ) {
+            event.setExecutionTime( nanoTime );
+        }
     }
 
 
     @Override
-    public PolyphenyDbSignature prepareQuery( RelRoot logicalRoot, RelDataType parameterRowType, boolean isRouted ) {
-        return prepareQuery( logicalRoot, parameterRowType, isRouted, false );
+    public RelRoot expandView( RelDataType rowType, String queryString, List<String> schemaPath, List<String> viewPath ) {
+        return null; // TODO
     }
```

Comment:
@cedricmendelin Whats with this todo?
Shouldn't this be addressed?

Reply:
Not coming from my side, this TODO is already on master branch available:
https://github.com/polypheny/Polypheny-DB/blob/master/dbms/src/main/java/org/polypheny/db/processing/AbstractQueryProcessor.java

Reply:
Will be removed as part of the language refactoring.